packages/radix-vue/src/FocusScope/FocusScope.vue

```diff
@@ -113,13 +113,18 @@ watchEffect((cleanupFn) => {
   // When the focused element gets removed from the DOM, browsers move focus
   // back to the document.body. In this case, we move focus to the container
   // to keep focus trapped correctly.
+  // -- related: https://github.com/radix-vue/radix-vue/issues/518
+  // Radix Vue tentative solution:
+  // instead of leaning on document.activeElement, we use lastFocusedElementRef.value to check
+  // if the element still exist inside the container,
+  // if not then we focus to the container
   function handleMutations(mutations: MutationRecord[]) {
-    const focusedElement = document.activeElement as HTMLElement | null
-    if (focusedElement !== document.body)
-      return
-    for (const mutation of mutations) {
-      if (mutation.removedNodes.length > 0)
-        focus(container)
+    const isLastFocusedElementExist = container.contains(lastFocusedElementRef.value)
+    if (!isLastFocusedElementExist) {
+      for (const mutation of mutations) {
```

Comment:
Do we need to loop through the mutations and check for removed nodes? 
`lastFocusedElementRef` it is always up to date anyway.

Just tested this: 
```ts
function handleMutations(mutations: MutationRecord[]) {
  const isLastFocusedElementExist = container.contains(lastFocusedElementRef.value)
  if (!isLastFocusedElementExist) focus(container)
}
```
and it handles focus properly when: 
- [x] focused element gets removed
- [x] some element added on button click
- [x] some element removed on button click
- [x] some element added on input change
- [x] some element removed on input change

Reply:
Ah make sense! Thanks for checking the use cases. I've tested too and it seems like working as expected!